### PR TITLE
Dont overwrite the subboard posttime when it's newer fixes #5017

### LINKED ISSUE #5017: root board overwrite sub board posttime
#### Description
The different Postime are not right evaluated

### Steps to reproduce
1.  create a board
2.  create a sub board
3. post something in the sub board
--> board index show number of post 1 but not the subject and poist

### Environment (complete as necessary)
- Version/Git revision: newest
- Database Type:
- Database Version:
- PHP Version:

### Additional information/references
https://www.simplemachines.org/community/index.php?topic=562333.msg3989296#new

### PR DESCRIPTION
**Pull request by albertlast**
issue: #5017

### CODE REVIEW

**Comment by albertlast** [comment]
The pr fixed the issue: https://www.simplemachines.org/community/index.php?topic=562333.msg3989423#msg3989423

beside of fixing the issue i align the cte query with the normal query, by droping the root board

**Comment by albertlast** [comment]
@Sesquipedalian this pr is done, i would recomand based on the issue, that this got included into rc1.

File: Sources/Subs-BoardIndex.php [hunk 1]
```diff
@@ -296,6 +297,26 @@ function getBoardIndex($boardIndexOptions)
 					$row_boards[$row_board['id_parent']]['num_posts'] += $row_board['num_posts'];
 					$row_boards[$row_board['id_parent']]['num_topics'] += $row_board['num_topics'];
 				}
+
+				if (forum_time(true, $row_boards[$row_board['id_parent']]['poster_time']) < forum_time(true, $row_board['poster_time']))
```
Comment (Sesquipedalian) [inline Sources/Subs-BoardIndex.php]: I don't think you need to call `forum_time()` here. Just use the raw poster_time values.

File: Sources/Subs-BoardIndex.php [hunk 2]
```diff
@@ -346,8 +367,18 @@ function getBoardIndex($boardIndexOptions)
 		}
 
 		// Set the last post in the parent board.
-		if ($row_board['id_parent'] == $boardIndexOptions['parent_id'] || ($isChild && !empty($row_board['poster_time']) && forum_time(true, $row_boards[$row_board['id_parent']]['poster_time']) < forum_time(true, $row_board['poster_time'])))
-			$this_category[$isChild ? $row_board['id_parent'] : $row_board['id_board']]['last_post'] = $this_last_post;
+		if ($isChild && !empty($row_board['poster_time']) 
+				&& forum_time(true, $row_boards[$row_board['id_parent']]['poster_time']) < forum_time(true, $row_board['poster_time']))
```
Comment (Sesquipedalian) [inline Sources/Subs-BoardIndex.php]: I don't think you need to call `forum_time()` here. Just use the raw poster_time values.

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]
This looks pretty good. There are two lines where you can remove some unnecessary calls to `forum_time()`, and then it'll be ready to merge.

**Comment by albertlast** [comment]
Fixed the unnessary calls

**Review by Sesquipedalian — APPROVED** [review]

**Pull request merged**